pyro/advection_ppm/simulation.py

```diff
@@ -0,0 +1,132 @@
+import importlib
```

Comment:
you should just subclass `advection` -- that way you don't need all of this code.  Take a look at how `advection_rk` does it.  In your case, I don't think you need to have any function overrides